.github/workflows/ci.yml

```diff
@@ -0,0 +1,112 @@
+name: 'ci'
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: 'Build on Node v${{ matrix.node }} ${{ matrix.os }} ${{ matrix.arch }}'
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        os: [Ubuntu, Windows, macOS]
+        arch: [x64]
+        node: ['12', '14', '16']
+        include:
+          - os: Windows
+            arch: x86
+            node: '12'
+          - os: Windows
+            arch: x86
+            node: '14'
+          - os: Windows
+            arch: x86
+            node: '16'
+
+    steps:
+      - name: 'Checkout repository'
+        uses: actions/checkout@v2
+
+      - name: 'Install Node.js v${{ matrix.node }} ${{ matrix.arch }}'
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+          architecture: ${{ matrix.arch }}
+
+      - name: 'Install Linux dependencies'
+        if: ${{ matrix.os == 'Ubuntu' }}
+        run: |
+          sudo apt-get update
+          sudo apt-get install libudev-dev
+
+      - name: 'Install dependencies and build from source'
+        run: npm ci --build-from-source
+
+      # TODO: Tests are disabled until we have tests which are
+      # suitable for CI and don't require manual interaction.
+      - name: 'Test'
+        run: echo "Skipping tests on CI, as they currently require manual interaction."
+
+      # Prebuilds should only be generated once per OS + arch
+      # Running from Node LTS will generate prebuilds for all ABIs
+      - name: 'Prebuild binaries for all ABIs'
+        if: ${{ matrix.node == '14' }}
+        run: npm run prebuild
```

Comment:
What's the point of having the whole matrix of `node: ['12', '14', '16']` if we can just build for all ABI's from Node v14?

Reply:
> A matrix of jobs to run npm ci --build-from-source on all the OS's and architectures we want

It looks like we use this so we can test building/installing the package